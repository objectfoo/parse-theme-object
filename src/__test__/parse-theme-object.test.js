const parseThemeObject = require('../../');
const data = require('./data/index');

describe('stuff', () => {
	it('should throw if called without object', () => {
		expect(() => {
			parseThemeObject();
		}).toThrow();
	});

	it('should parse provided object', () => {
		const r = parseThemeObject(data.simple);
		expect(r).toMatchSnapshot();
	});

	it('should support optional start path as string', () => {
		const r = parseThemeObject(data.simple, {
			start: 'start'
		});
		expect(r).toMatchSnapshot();
	});

	it('should support optional start path as array', () => {
		const r = parseThemeObject(data.simple, {
			start: ['start']
		});
		expect(r).toMatchSnapshot();
	});

	it('should throw if start node not found', () => {
		expect(() => {
			parseThemeObject(data.simple, {
				start: ['fakekey']
			});
		}).toThrow();
	});

	it('should blacklist properties', () => {
		const r = parseThemeObject(data.blacklist, {
			blacklist: ['blacklistme']
		});
		expect(r).toMatchSnapshot();
	});

	it('should ignore functions', () => {
		const r = parseThemeObject(data.function);
		expect(r).toMatchSnapshot();
	});

	it('should parse nested sections', () => {
		const r = parseThemeObject(data.full, {
			start: ['palette'],
			blacklist: ['contrastDefaultColor']
		});
		expect(r).toMatchSnapshot();
	});
});
