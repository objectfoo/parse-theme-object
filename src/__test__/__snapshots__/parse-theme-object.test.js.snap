// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stuff should blacklist properties 1`] = `
Array [
  Object {
    "path": Array [
      "a",
    ],
    "value": "a",
  },
]
`;

exports[`stuff should ignore functions 1`] = `
Array [
  Object {
    "path": Array [
      "a",
    ],
    "value": "a",
  },
]
`;

exports[`stuff should parse nested sections 1`] = `
Array [
  Object {
    "path": Array [
      "type",
    ],
    "value": "light",
  },
  Object {
    "header": true,
    "path": Array [
      "common",
    ],
  },
  Object {
    "path": Array [
      "common",
      "black",
    ],
    "value": "#000",
  },
  Object {
    "path": Array [
      "common",
      "white",
    ],
    "value": "#fff",
  },
  Object {
    "path": Array [
      "common",
      "transparent",
    ],
    "value": "rgba(0, 0, 0, 0)",
  },
  Object {
    "path": Array [
      "common",
      "fullBlack",
    ],
    "value": "rgba(0, 0, 0, 1)",
  },
  Object {
    "header": true,
    "path": Array [
      "primary",
    ],
  },
  Object {
    "path": Array [
      "primary",
      "50",
    ],
    "value": "#prim50",
  },
  Object {
    "path": Array [
      "primary",
      "100",
    ],
    "value": "#prim100",
  },
  Object {
    "header": true,
    "path": Array [
      "secondary",
    ],
  },
  Object {
    "path": Array [
      "secondary",
      "50",
    ],
    "value": "#seco50",
  },
  Object {
    "path": Array [
      "secondary",
      "100",
    ],
    "value": "#seco100",
  },
  Object {
    "header": true,
    "path": Array [
      "shades",
    ],
  },
  Object {
    "header": true,
    "path": Array [
      "shades",
      "dark",
    ],
  },
  Object {
    "header": true,
    "path": Array [
      "shades",
      "dark",
      "text",
    ],
  },
  Object {
    "path": Array [
      "shades",
      "dark",
      "text",
      "primary",
    ],
    "value": "rgba(255, 255, 255, 1)",
  },
  Object {
    "path": Array [
      "shades",
      "dark",
      "text",
      "secondary",
    ],
    "value": "rgba(255, 255, 255, 0.7)",
  },
]
`;

exports[`stuff should parse provided object 1`] = `
Array [
  Object {
    "path": Array [
      "a",
    ],
    "value": "a",
  },
  Object {
    "header": true,
    "path": Array [
      "start",
    ],
  },
  Object {
    "path": Array [
      "start",
      "b",
    ],
    "value": "b",
  },
]
`;

exports[`stuff should support optional start path as array 1`] = `
Array [
  Object {
    "path": Array [
      "b",
    ],
    "value": "b",
  },
]
`;

exports[`stuff should support optional start path as string 1`] = `
Array [
  Object {
    "path": Array [
      "b",
    ],
    "value": "b",
  },
]
`;
